serial_protocol.md

```diff
@@ -186,3 +192,135 @@ Used to stop listening to events at a given path. This must be the same path pro
 	>>BEGIN_STREAM /user/aturing
 	>>END_STREAM /user/aturing
 	<<+OK
+
+#Messaging Protocol
+
+## BEGIN_MESSAGING 
+Called to start communicating with Firebase Cloud Messaging, it requires a server key.
+###Usage
+	BEGIN %Server_Key%
+###Response
+	+OK - Accepted initialization parameters
+###Examples
+	>> BEGIN https://samplechat.firebaseio-demo.com
+	<< +OK
+
+## MESSAGE_TO_USERS
+Called to start composing a message to users by registration id. You can specify as many as you wish separated by spaces on the same line. 
```

Comment:
I did but there are a bunch of options. It is basically this struct we are defining: https://github.com/googlesamples/firebase-arduino/blob/master/src/FirebaseCloudMessaging.h#L34

So we could have a bunch of variations of single line commands, or one command with some very detailed syntax, or one command with some kind of argument system. Like ```MSG REGISTRATIONID high_priority=true collapse_key=current_temp KEY VALUE KEY VALUE..```

Or just create a stateful way like this. This emulates the SMTP protocol which has a very similar workflow.